apps/studio/components/interfaces/DataWarehouse/WarehouseCollectionDetail.tsx

```diff
@@ -0,0 +1,168 @@
+import LoadingOpacity from 'components/ui/LoadingOpacity'
+import ShimmerLine from 'components/ui/ShimmerLine'
+import { useRouter } from 'next/router'
+import React, { useEffect } from 'react'
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  IconRefreshCcw,
+  IconRewind,
+} from 'ui'
+import { LogTable } from '../Settings/Logs'
+import { useWarehouseQueryQuery } from 'data/analytics/warehouse-query'
+import { useWarehouseCollectionsQuery } from 'data/analytics/warehouse-collections-query'
+import Link from 'next/link'
+import { useWarehouseAccessTokensQuery } from 'data/analytics/warehouse-access-tokens-query'
+import { TestCollectionDialog } from './TestCollectionDialog'
+
+export const WarehouseCollectionDetail = () => {
+  const router = useRouter()
+  const collectionToken = router.query.collectionToken as string
+  const projectRef = router.query.ref as string
+  const accessTokens = useWarehouseAccessTokensQuery({ projectRef })
+
+  const notProd = process.env.NODE_ENV !== 'production'
+
+  const { data: collections, isLoading: collectionsLoading } = useWarehouseCollectionsQuery(
+    { projectRef },
+    { enabled: true }
+  )
+  const collection = (collections || []).find((c) => c.token === collectionToken)
+  const [params, setParams] = React.useState({
+    sql: `select current_timestamp() as 'time'`,
+  })
+
+  const [pagination, setPagination] = React.useState({
+    limit: 100,
+    offset: 0,
+  })
+
+  useEffect(() => {
+    if (collection) {
+      setParams({
+        ...params,
+        sql: `
+        select id, timestamp, event_message from \`${collection.name}\` 
+        where timestamp > '2024-01-01' 
+        order by timestamp desc limit ${pagination.limit} offset ${pagination.offset}
+        `,
+      })
+    }
+  }, [collection, pagination])
+
+  const {
+    isLoading: queryLoading,
+    data: queryData,
+    isError,
+    refetch,
+    isRefetching,
+  } = useWarehouseQueryQuery({ ref: projectRef, sql: params.sql }, { enabled: !!params.sql })
+
+  const formatResults = (results: any) => {
+    if (!results || !results.length) {
+      return []
+    }
+
+    const r = results.map(({ timestamp, ...r }: any) => {
+      return {
+        timestamp: new Date(timestamp / 1000).toLocaleString(),
+        ...r,
+      }
+    })
+
+    return r
+  }
+
+  const results = formatResults(queryData?.data?.result)
+
+  function loadOlder() {
```

Comment:
technically the Load older button is more of a "Load more" button, since it functions more as pagination.
Can also display the total number of rows rendered 👀